Updated readme file with better design, less text

ServiAmirPM opened the pull request:
Still requires more modeifications (results, features, tutorials)

## Pull Request Description:

- Re-structure - changed the way the document is structured (table of content), including changing titles names
- Redesigned the header - with better look and feel, with as much information in the header.
- Removed tutorials from the tutorials section in the readme - will add some back after re-structuring the tutorials.
- Added visuals to the page for better look and feel, and to better explain the product.
- improved the quantization flow table and added 4 tutorials to it.
- Re-ordered the Resources section, making it clear that API is per quantization method
- Hidden (collapsed) the supported version table. users can open it on demand
- Supported features - didn't touch it - still needs work
- Quantization - didn't touch it - still needs work
- Results - removed text, still requires updating the numbers, add mixed percision and the update graph
- Toubleshooting and community - added community to this section.
- Updated the license section with text and a badge

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

ofirgo commented:
Please provide more details regarding the changes in the PR description (what are the main changes to the readme, what was added/removed, etc.), both for documentation and to make my review easier 😄

ServiAmirPM commented:
- Re-structure - changed the way the document is structured (table of content), including changing titles names 
- Redesigned the header - with better look and feel, with as much information in the header.
- Removed tutorials from the tutorials section in the readme - will add some back after re-structuring the tutorials.
- Added visuals to the page for better look and feel, and to better explain the product.
- improved the quantization flow table and added 4 tutorials to it.
- Re-ordered the Resources section, making it clear that API is per quantization method
-  Hidden (collapsed) the supported version table. users can open it on demand
- Supported features - didn't touch it - still needs work
- Quantization - didn't touch it - still needs work
- Results - removed text, still requires updating the numbers, add mixed percision and the update graph
- Toubleshooting and community - added community to this section. 
- Updated the license section with text and a badge

ofirgo commented on README.md:
```diff
@@ -1,55 +1,104 @@
 
 Currently, MCT is being tested on various Python, Pytorch and TensorFlow versions:
-
+<details id="supported-versions">
```
I believe you have a conflict here with the current readme - we added Python 3.12 and Pytorch 2.5 to this table.
Maybe remove torch 2.1 so it won't be too much (also remove python 3.9 from the table?)
Let me know if you need help merging and solving the conflicts

ofirgo commented on README.md:
```diff
@@ -83,7 +133,7 @@ MCT supports different quantization methods:
 | PTQ                                           | Low        | Low (order of minutes)      |
-| GPTQ (parameters fine-tuning using gradients) | Mild       | Mild (order of 2-3 hours)   |
+| GPTQ (parameters fine-tuning using gradients) | Moderate   | Moderate (order of 2-3 hours)   |
 | QAT                                           | High       | High (order of 12-36 hours) |
```
the "Main features:" part below is unreadable, let's think how we can improve it

ServiAmirPM replied:
Agree - will fix in next round.

ofirgo commented on README.md:
```diff
@@ -83,7 +133,7 @@ MCT supports different quantization methods:
-| GPTQ (parameters fine-tuning using gradients) | Mild       | Mild (order of 2-3 hours)   |
+| GPTQ (parameters fine-tuning using gradients) | Moderate   | Moderate (order of 2-3 hours)   |
 | QAT                                           | High       | High (order of 12-36 hours) |
 
```
Also, need to fix the reference and title for the EPTQ paper

ofirgo commented on README.md:
```diff
@@ -148,6 +191,14 @@ In the following table we present the ImageNet validation results for these mode
+Below is a graph of [MobileNetV2](https://keras.io/api/applications/mobilenet/) accuracy on ImageNet vs average bit-width of weights (X-axis), using 
+single-precision quantization, mixed-precision quantization, and mixed-precision quantization with GPTQ. 
+
+<img src="https://github.com/sony/model_optimization/raw/main/docsrc/images/mbv2_accuracy_graph.png">
```
we are preparing this results figure in Pytorch, so need to think what are we going to include under the keras section results

ServiAmirPM replied:
Agree - next round.

ofirgo commented on README.md:
```diff
@@ -148,6 +191,14 @@ In the following table we present the ImageNet validation results for these mode
+
+<img src="https://github.com/sony/model_optimization/raw/main/docsrc/images/mbv2_accuracy_graph.png">
+
+For more results, please see [1]
```
This is a very old not very relevant reference (for results purposes).
If we need more results maybe we can try to create a separate table and reference to it, but I'm not sure that the results in this paper are representing for the current performance of MCT

ofirgo commented on README.md:
```diff
@@ -158,23 +209,25 @@ Results for applying pruning to reduce the parameters of the following models by
-## Trouble Shooting
+## Troubleshooting and Community
 
 If the accuracy degradation of the quantized model is too large for your application, check out the [Quantization Troubleshooting](https://github.com/sony/model_optimization/tree/main/quantization_troubleshooting.md)
```
"If the accuracy degradation of the quantized model is too large for your application" -->
"If you encountered large accuracy degradation from the quantized model returned by MCT..."

ServiAmirPM replied:
fixed

ofirgo commented on README.md:
```diff
@@ -158,23 +209,25 @@ Results for applying pruning to reduce the parameters of the following models by
 MCT aims at keeping a more up-to-date fork and welcomes contributions from anyone.
 
-*You will find more information about contributions in the [Contribution guide](https://github.com/sony/model_optimization/blob/main/CONTRIBUTING.md).
+*Checkout more our [Contribution guide](https://github.com/sony/model_optimization/blob/main/CONTRIBUTING.md) for more details.
```
remove "more"

ServiAmirPM replied:
fixed

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
+  <a href="#license">License</a>
+</p>
+<p align="center">
+  <a href="https://sony.github.io/model_optimization#prerequisites"><img src="https://img.shields.io/badge/python-2.1%20%7C%202.2%20%7C%202.3-blue" /></a>
```
Solved already

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
-- [Troubleshooting](https://github.com/sony/model_optimization?tab=readme-ov-file#trouble-shooting)
-- [Contributions](https://github.com/sony/model_optimization?tab=readme-ov-file#contributions)
-- [License](https://github.com/sony/model_optimization?tab=readme-ov-file#license)
+Our [tutorials](https://github.com/sony/model_optimization/blob/main/tutorials/README.md) section will walk you through the basics of deploying your first model; Covering various compression techniques for both Keras and PyTorch models. 
```
Done. Committed fix.

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
+<p align="center">
 
-## Getting Started
+  Quantization Method  | Complexity | Computational Cost | Tutorial 
```
Resolved. removed the grey ones.

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
+  Quantization Method  | Complexity | Computational Cost | Tutorial 
+-------------------- | -----------|--------------------|---------
+PTQ (Post Training Quantization)  | Low | Low (~1-10 CPU minutes) | <a href="https://colab.research.google.com/github/sony/model_optimization/blob/main/tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_post_training_quantization.ipynb"><img src="https://img.shields.io/badge/Pytorch-green"/></a> <a href="https://colab.research.google.com/github/sony/model_optimization/blob/main/tu ...
+GPTQ (parameters fine-tuning using gradients)  | Moderate | Moderate (~2-3 GPU hours) | <a href="https://colab.research.google.com/github/sony/model_optimization/blob/main/tutorials/notebooks/mct_features_notebooks/keras/example_keras_mobilenet_gptq.ipynb"><img src="https://img.shields.io/badge/Keras-green"/></a> <img src="https://img.shields.io/badge/Pytorch-grey"/>
```
fixed

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
 
-### Installation
+For each flow, **Quantization core** utilizes various algorithms and hyper-parameters for optimal [hardware-aware](https://github.com/sony/model_optimization/blob/main/model_compression_toolkit/target_platform_capabilities/README.md) quantization results. 
+For further details, please see [Supported features and algorithms](#supported-features). User can either provide their own representative dataset, or utilize the [Data Generation](#data-generation-) capability.
```
Resolved

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
+<p align="center">
 
-For installing the nightly version or installing from source, refer to the [installation guide](https://github.com/sony/model_optimization/blob/main/INSTALLATION.md).
+<img src="/docsrc/images/mctFlow.png" width="800">
```
Will upgrade in upcoming future (designer will provide better one)

ServiAmirPM commented on README.md:
```diff
@@ -1,55 +1,104 @@
+* [User Guide](https://sony.github.io/model_optimization/docs/index.html)  contains detailed information about MCT and guides you from installation through optimizing models for your edge AI applications.
 
-### Quick start & tutorials 
+* MCT's [API Docs](https://sony.github.io/model_optimization/docs/api/api_docs/) is seperated per quantization methods:  
```
fixed
